### PR TITLE
Nadai/update develop v3

### PR DESCRIPTION
**Pull request by Nadai2010**
# Update develop V3 

Update to the Marquis with latest starknet react version, and all sync with Scaffold v3.

## Types of change

- [x] Feature
- [x] Bug
- [x] Enhancement

### CODE REVIEW

**Comment by gianalarcon** [comment]
@Nadai2010 Please leave your comments based on your PR. Thanks!

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,288 +0,0 @@
-name: Build and deploy
```
Comment (gianalarcon) [inline .github/workflows/release.yaml:1]: No need to delete this file @Nadai2010

File: packages/nextjs/app/brankit/page.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
+"use client";
```
Comment (gianalarcon) [inline packages/nextjs/app/brankit/page.tsx:1]: This file is already on develop branch. I dont expect to see this change. Let's check again of the PR is synced with develop branch

**Review by gianalarcon — CHANGES_REQUESTED** [review]

File: packages/nextjs/app/brankit/page.tsx (hunk 2 above)
Comment (Nadai2010) [reply packages/nextjs/app/brankit/page.tsx:1]: Oops sorry, that was a mistake in the folder name, I will correct it now and upload important comments so @Buuh2511 can review it better.

File: packages/nextjs/components/SwitchTheme.tsx [hunk 3]
```diff
@@ -7,7 +7,6 @@ import { MoonIcon, SunIcon } from "@heroicons/react/24/outline";
 export const SwitchTheme = ({ className }: { className?: string }) => {
   const { setTheme, resolvedTheme } = useTheme();
   const [mounted, setMounted] = useState(false);
-  console.log("resolvedTheme", resolvedTheme);
```
Comment (Nadai2010) [inline packages/nextjs/components/SwitchTheme.tsx:10]: V3 Integration, (remove unused logs and comments)

File: packages/nextjs/components/scaffold-stark/Balance.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/Balance.tsx:1]: V3 Integration

File: packages/nextjs/components/scaffold-stark/BlockExplorer.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/BlockExplorer.tsx:1]: V3 Integration

File: packages/nextjs/components/scaffold-stark/BlockExplorerSepolia.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/BlockExplorerSepolia.tsx:1]: V3 Integration

File: packages/nextjs/components/scaffold-stark/ClassHash.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/ClassHash.tsx:1]: V3 Integration, now we have a component that controls the clash hash obtained.

File: packages/nextjs/components/scaffold-stark/CustomConnectButton/AddressInfoDropdown.tsx [hunk 4]
```diff
@@ -37,6 +37,10 @@ export const AddressInfoDropdown = ({
   const { resolvedTheme } = useTheme();
   // const isDarkMode = resolvedTheme === "dark";
   const [openWalletModal, setOpenWalletModal] = useState(false);
+  const iconSrc =
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/CustomConnectButton/AddressInfoDropdown.tsx:40]: Idem, type in icon

File: packages/nextjs/components/scaffold-stark/CustomConnectButton/NetworkOptions.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/CustomConnectButton/NetworkOptions.tsx:1]: V3 integration

File: packages/nextjs/components/scaffold-stark/CustomConnectButton/Wallet.tsx [hunk 5]
```diff
@@ -37,13 +39,22 @@ const Wallet = ({
                 __html: connector.icon.light ?? "",
               }}
             />
-          ) : (
+          ) : typeof connector.icon === "object" ? (
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/CustomConnectButton/Wallet.tsx:42]: Idem, type in icon

File: packages/nextjs/components/scaffold-stark/Faucet.tsx
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/Faucet.tsx:1]: Faucet update in Scaffold Stark, migration without changes.

File: packages/nextjs/components/scaffold-stark/FaucetButton.tsx [hunk 6]
```diff
@@ -4,9 +4,9 @@ import { useState } from "react";
 import { BanknotesIcon } from "@heroicons/react/24/outline";
 import { mintEth } from "~~/services/web3/faucet";
 import { Address, devnet } from "@starknet-react/chains";
-import { useAccount } from "@starknet-react/core";
 import { useTargetNetwork } from "~~/hooks/scaffold-stark/useTargetNetwork";
 import useScaffoldEthBalance from "~~/hooks/scaffold-stark/useScaffoldEthBalance";
+import { useAccount } from "~~/hooks/useAccount";
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/FaucetButton.tsx:9]: import `useAccount` now we use it from hooks/useAccount

File: packages/nextjs/hooks/scaffold-stark/useAutoConnect.ts [hunk 7]
```diff
@@ -12,22 +12,31 @@ export const useAutoConnect = (): void => {
...
-      );
-      if (connector) {
-        if (
-          connector.id == "burner-wallet" &&
-          savedConnector?.ix !== undefined
-        ) {
-          (connector as BurnerConnector).burnerAccount =
-            burnerAccounts[savedConnector.ix];
+      const currentTime = Date.now();
+      const ttlExpired =
```
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useAutoConnect.ts:23]: New useAutoConnect hooks with some changes like TTL improvements.

File: packages/nextjs/hooks/scaffold-stark/useScaffoldEthBalance.ts [hunk 8]
```diff
@@ -12,8 +12,8 @@ type UseScaffoldEthBalanceProps = {
 const useScaffoldEthBalance = ({ address }: UseScaffoldEthBalanceProps) => {
   const { data: deployedContract } = useDeployedContractInfo("Eth");
 
-  const { data, ...props } = useContractRead({
-    functionName: "balanceOf",
+  const { data, ...props } = useReadContract({
+    functionName: "balance_of",
```
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useScaffoldEthBalance.ts:16]: In addition to the new `useReadContract`, we also use `balance_of`

File: packages/nextjs/hooks/useAccount.ts
Comment (Nadai2010) [inline packages/nextjs/hooks/useAccount.ts:1]: V3 Migration, New hook useAccount

File: packages/nextjs/scaffold.config.ts [hunk 9]
```diff
@@ -6,6 +6,7 @@ export type ScaffoldConfig = {
   onlyLocalBurnerWallet: boolean;
   rpcProviderUrl: string;
   walletAutoConnect: boolean;
+  autoConnectTTL: number;
```
Comment (Nadai2010) [inline packages/nextjs/scaffold.config.ts:9]: New feature the autoConnect (TTL is configure in 60000 ms)

File: packages/nextjs/services/web3/connectors.tsx
Comment (Nadai2010) [inline packages/nextjs/services/web3/connectors.tsx:1]: In migration V3 add workaround on wallet connectors to remove from localStorage

File: packages/snfoundry/package.json [hunk 10]
```diff
@@ -4,11 +4,14 @@
...
     "deploy": "ts-node scripts-ts/helpers/deploy-wrapper.ts",
+    "deploy:reset": "ts-node scripts-ts/helpers/deploy-wrapper.ts --reset",
     "test": "cd contracts && snforge test",
     "test-eslint": "node eslint-contract-name/eslint-plugin-contract-names.test.js",
     "compile": "cd contracts && scarb build",
     "format": "prettier --write 'scripts-ts/**/*.{ts,tsx}' && cd contracts && scarb fmt",
-    "format:check": "prettier --check 'scripts-ts/**/*.{ts,tsx}' && cd contracts && scarb fmt --check"
+    "format:check": "prettier --check 'scripts-ts/**/*.{ts,tsx}' && cd contracts && scarb fmt --check",
+    "postinstall": "shx cp -n .env.example .env",
+    "verify": "ts-node scripts-ts/verify-contracts.ts"
```
Comment (Nadai2010) [inline packages/snfoundry/package.json:14]: Now we can verify our SC with walnut, we'll use for example `yarn verify --network sepolia`

File: packages/snfoundry/scripts-ts/helpers/deploy-wrapper.ts [hunk 11]
```diff
@@ -6,22 +6,32 @@ interface CommandLineOptions {
   _: string[]; // Non-hyphenated arguments are usually under the `_` key
   $0: string; // The script name or path is under the `$0` key
   network?: string; // The --network option
+  reset?: boolean;
+  fee?: string;
```
Comment (Nadai2010) [inline packages/snfoundry/scripts-ts/helpers/deploy-wrapper.ts:10]: Now we can deploy our SC using STRK for the fee, we will have to indicate the command `--fee strk` for example, `yarn deploy --fee strk`

File: packages/snfoundry/scripts-ts/verify-contracts.ts
Comment (Nadai2010) [inline packages/snfoundry/scripts-ts/verify-contracts.ts:1]: New script the verify SC

**Pull request closed without merging**